From 601c8c40dd8e17083b060418bb630c87e168be86 Mon Sep 17 00:00:00 2001
From: Colin Cross <ccross@android.com>
Date: Mon, 13 Jun 2011 00:45:59 +0000
Subject: ARM: gic: Allow gic arch extensions to provide irqchip flags

Tegra can benefit from the IRQCHIP_MASK_ON_SUSPEND flag, allow it
to be passed to the gic irq chip.

Signed-off-by: Colin Cross <ccross@android.com>
Signed-off-by: Santosh Shilimkar <santosh.shilimkar@ti.com>
Reviewed-by: Kevin Hilman <khilman@ti.com>
Tested-and-Acked-by: Shawn Guo <shawn.guo@linaro.org>
Tested-by: Vishwanath BS <vishwanath.bs@ti.com>
(cherry picked from commit 9c12845ee49716209cb2b087c0b47c3e37096bde)

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 arch/arm/common/gic.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/common/gic.c b/arch/arm/common/gic.c
index 66c7c48..734db99 100644
--- a/arch/arm/common/gic.c
+++ b/arch/arm/common/gic.c
@@ -544,6 +544,7 @@ void __init gic_init(unsigned int gic_nr, unsigned int irq_start,
 	if (gic_nr == 0)
 		gic_cpu_base_addr = cpu_base;
 
+	gic_chip.flags |= gic_arch_extn.flags;
 	gic_dist_init(gic, irq_start);
 	gic_cpu_init(gic);
 	gic_pm_init(gic);
-- 
1.7.10.2.565.gbd578b5

